### ISSUE TITLE
whereDate() does not wrap comparison value with DATE()

### ISSUE THREAD

**Issue by troygilbert** [laravel/framework#25311]
This may be considered more of a feature request, but I've run into enough folks who had the same assumptions as me that I thought I'd raise it as a possible bug.

While `whereDate()` does wrap the column with `DATE()`, it does not wrap the _value_ with `DATE()`. This results in scenarios where you pass a `Carbon` instance as the value and it gets converted to the string `"2018-08-23 12:35:00"` which will never satisfy a where clause with a column wrapped in `DATE()`. This behavior forces code to be littered with `$carbonInstance->toDateString()`, which just doesn't feel like the Laravel Way™. ;-)

Put simply, I assumed `whereDate('column_name', $value)` was equivalent to this:
```
whereRaw('DATE(column_name) = DATE(?)', $value);
```
But it's actually equivalent to this:
```
whereRaw('DATE(column_name) = ?', $value);
```

(Along similar lines, I'd assume the comparison value would be wrapped with appropriate functions for `whereMonth()`, `whereDay()`, and `whereYear()`.)

**Comment by dwightwatson** [comment]
This snagged me the other day, had to call `toDateString()` on the Carbon instance. Feel like it would be something nice to have handled under the hood. 

I wonder if it would be better to wrap the value in the same `DATE` function or just format it appropriately.

**Comment by decadence** [comment]
@troygilbert 
Merged, can be closed
